blocks/components/editable/index.js

```diff
@@ -127,6 +134,19 @@ export default class Editable extends wp.element.Component {
 		}
 	}
 
+	componentWillReceiveProps( nextProps ) {
+		const formatMap = {
+			strong: 'bold',
+			em: 'italic'
+		};
+
+		forEach( nextProps.formats, ( state, format ) => {
+			if ( state !== this.props.formats[ format ] ) {
+				this.editor.formatter.toggle( formatMap[ format ] );
```

Comment:
Or if `toggle` is the right way to do it, we should probably check whether the current path is already equivalent to the `this.props.formats[ format ]` value before calling `toggle`